src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif

```diff
@@ -74,42 +82,153 @@
           "formattedRuleMessage": {
             "formatId": "default",
             "arguments": [
-              "n"
+              "p"
             ]
           },
+          "suppressionStates": [ "suppressedExternally" ],
+          "baselineState": "existing",
+          "level": "error",
+          "snippet": "add_core(p, offset, val);\n    return;",
           "locations": [
             {
               "analysisTarget": {
-                "uri": "collections/list.cpp",
+                "uri": "file:///home/build/src/collections/list.cpp"
+              },
+              "resultFile": {
+                "uri": "file:///home/build/src/collections/list.h",
                 "region": {
-                  "startLine": 15
+                  "startLine": 15,
+                  "startColumn": 9,
+                  "endLine": 15,
+                  "endColumn": 10,
+                  "length": 1,
+                  "offset": 254
+                }
+              },
+              "fullyQualifiedLogicalName": "collections::list:add",
+              "decoratedName": "?add@list@collections@@QAEXH@Z"
+            }
+          ],
+          "relatedLocations": [
+            {
+              "message": "\"count\" was declared here.",
+              "physicalLocation": {
+                "uri": "file:///home/build/src/collections/list.h",
+                "region": {
+                  "startLine": 8,
+                  "startColumn": 5
                 }
               },
               "fullyQualifiedLogicalName": "collections::list:add"
             }
           ],
           "codeFlows": [
             {
-              "message": "Code flow #1",
+              "message": "Path from declaration to usage",
               "locations": [
                 {
                   "step": 0,
                   "kind": "declaration",
                   "importance": "essential",
-                  "fullyQualifiedLogicalName": "collections::list::add",
+                  "message": "Variable \"p\" declared.",
```

Comment:
I don't agree with this. My concern is less about making the log file readable and more about making the messages displayed to the end user be grammatical. There's nothing in the Microsoft Manual of Style that recommends single quotes, and Chicago is specific about double quotes except in certain cases (British English is one of them). I'm going to stick with double quotes unless you have a citation that specifies single quotes. (I know single quotes are widely used in MS error messages, probably for the reason you cite (readabilty of the message in code), but I've never agreed with it.)

---
In reply to: [72280385](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72280385) [](ancestors = 72280385)